apps/web/src/hooks/useRoutes.ts

```diff
@@ -0,0 +1,19 @@
+import { useMemo } from 'react';
+import { useLocation } from 'react-router-dom';
+import { PUBLIC_ROUTES_PREFIXES } from '../constants/routes';
+
+export function useRoutes() {
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
✏ **suggestion:** Please consider a more closely-scoped name; this feels overly generic. Perhaps `useRouteScopes`?‏

Reply:
Done!